src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -96,80 +96,102 @@ protected Task<ImmutableArray<CodeActionOperation>> GetPreviewOperationsAsync(Ca
 
         public void Invoke(CancellationToken cancellationToken)
```

Comment:
@AmadeusW was there some guarantee that once we implement ISuggestedAction3 the other Invoke methods won't get called anymore? I'd greatly simplify this if we could just delete these implementations.

Reply:
1. Yes, that would be nice.  
2. Even if VS guarantees that, do we need to worry about other consumers (like 3rd parties) calling into this?  I'm fine saying we don't support that scenario anymore.

Reply:
preemptively deprecating the original codepath for now and moving to just the IUIThreadOperationContext version.  Will add back in if amadeus says it's necessary.